### PR TITLE
Implement CCSocialManager::isAvailable on Android.

### PR DESCRIPTION
**Pull request by yalab**
Hi.
I implement CCSocialManager::isAvaiable on android.
Thanks.

### CODE REVIEW

File: CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java [hunk 1]
```diff
@@ -105,4 +108,25 @@ static public void onActivityResult(int requestCode, int resultCode, Intent inte
         }
         executeCallback(PostResult.SUCCEED.ordinal());
     }
+
+    static public boolean isTwitterAvailable () {
```
Comment (giginet) [inline CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java]: spaces before `()` is not needed.

`static public boolean isTwitterAvailable()`

File: CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java [hunk 2]
```diff
@@ -105,4 +108,25 @@ static public void onActivityResult(int requestCode, int resultCode, Intent inte
...
+        return isAvailable(FACEBOOK_PACKAGE_NAME);
+    }
+
+    static private boolean isAvailable (String packageName) {
+        Context context = Cocos2dxActivity.getContext();
+        PackageManager packageManager = context.getPackageManager();
+        List<ApplicationInfo> appInfoList = packageManager.getInstalledApplications(PackageManager.GET_META_DATA);
+
+        for (ApplicationInfo info : appInfoList) {
+          if (packageName.equals(info.processName)) {
```
Comment (giginet) [inline CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java]: indent is wrong 🤔

File: CCSocialShare/android/CCSocialManager.cpp [hunk 3]
```diff
@@ -13,7 +14,23 @@ namespace CCSocialShare {
...
-        return true;
+        JniMethodInfo methodInfo;
+        std::string methodName;
+        if (service == Service::TWITTER) {
+            methodName = "isTwitterAvailable";
+        } else if (service == Service::FACEBOOK) {
+            methodName = "isFacebookAvailable";
+        }
+        if(JniHelper::getStaticMethodInfo(methodInfo, SOCIAL_SHARE_CLASS_NAME, methodName.c_str(), "()Z"))
+        {
```
Comment (giginet) [inline CCSocialShare/android/CCSocialManager.cpp]: linebreak is not needed

File: CCSocialShare/android/CCSocialManager.cpp [hunk 4]
```diff
@@ -13,7 +14,23 @@ namespace CCSocialShare {
...
+            JNIEnv* env = JniHelper::getEnv();
```
Comment (giginet) [inline CCSocialShare/android/CCSocialManager.cpp]: `*` should be with variable name.

`JNIEnv *env`

File: CCSocialShare/android/CCSocialManager.cpp [hunk 5]
```diff
@@ -13,7 +14,23 @@ namespace CCSocialShare {
...
+            jboolean jbool = methodInfo.env->CallStaticBooleanMethod(methodInfo.classID, methodInfo.methodID);
+            methodInfo.env->DeleteLocalRef(methodInfo.classID);
+            bool b = (jbool == JNI_TRUE);
```
Comment (giginet) [inline CCSocialShare/android/CCSocialManager.cpp]: variable `b` is not needed

`return jbool == JNI_TRUE`

File: CCSocialShare/android/CCSocialManager.cpp [hunk 6]
```diff
@@ -13,7 +14,23 @@ namespace CCSocialShare {
...
+            return b;
+        }else{
```
Comment (giginet) [inline CCSocialShare/android/CCSocialManager.cpp]: `} else {`

File: CCSocialShare/android/CCSocialManager.cpp [hunk 7]
```diff
@@ -13,7 +14,23 @@ namespace CCSocialShare {
...
     {
-        return true;
+        JniMethodInfo methodInfo;
+        std::string methodName;
+        if (service == Service::TWITTER) {
+            methodName = "isTwitterAvailable";
+        } else if (service == Service::FACEBOOK) {
+            methodName = "isFacebookAvailable";
+        }
+        if(JniHelper::getStaticMethodInfo(methodInfo, SOCIAL_SHARE_CLASS_NAME, methodName.c_str(), "()Z"))
```
Comment (giginet) [inline CCSocialShare/android/CCSocialManager.cpp]: `if (`

**Review by giginet — CHANGES_REQUESTED** [review]
@yalab 
Thanks for your contribution.
I commented some syntax issues.
Could you fix it?

**Review by giginet — APPROVED** [review]
Thank you!
I'll merge and release newer version 🚀

**Comment by yalab** [comment]
@giginet 
I fixed about your comment.

**Pull request merged**